src/libraries/System.Private.CoreLib/src/System/String.cs

```diff
@@ -320,6 +320,7 @@ private static unsafe string Ctor(ReadOnlySpan<char> value)
         }
 
         public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)
+            where TState : allows ref struct
```

Comment:
this is super nice

Reply:
How come this in particular is so nice?

Reply:
Because it's reasonably common to have a span you want to pass in, and before this change that requires hackiness, e.g.
https://github.com/dotnet/runtime/pull/102795/files#diff-f2ce8db038c1b2e73c10ba711fb96c70c3777da9e18363e8c03c9300c032ee3bL15